homeassistant/components/deconz/deconz_device.py

```diff
@@ -72,3 +79,21 @@ def device_info(self):
             'sw_version': self._device.swversion,
             'via_hub': (DECONZ_DOMAIN, bridgeid),
         }
+
+    async def async_device_set_state(self, data, tries=0):
+        """Set state of device."""
+        from pydeconz import errors
+
+        try:
+            await self._device.async_set_state(data)
+
+        except errors.BridgeBusy:
+            async def retry_set_state(_now):
+                """Retry set state."""
+                await self.async_device_set_state(data, tries + 1)
+
+            if tries < 5:
+                retry_delay = 2 ** (tries + 1)
+                self._cancel_retry_set_state = \
+                    self.hass.helpers.event.async_call_later(
+                        retry_delay, retry_set_state)
```

Comment:
Do you think that is a good idea? What is going on if you set a new state after it is already one scheduled?

I love to see such retry logic more inside the library. The question is mostly, what is really needed and what can be worse in some situation.

Reply:
This doesn't really happen that often. Normal times this occurs is if you do an all lights on/off action, which should mostly happen when turning off all lights for the night. You don't do additional controls to the device in those cases so I am not really worried about any weird behaviours here.

Isn't this a normal abstraction letting hass do the retries?

Reply:
Moved retry mechanism to library